### ISSUE TITLE
MobileOneSignal cannot be used with C3s GooglePlay Plugin

### ISSUE THREAD

**Issue by Boulerzzz** [Chadori/construct-master-collection#217]
## Request Type
This is a **Bug Report**.

## Describe the bug
When adding C3s GooglePlay plugin and MobileOneSignal the build fails.

## Reproducible .c3p or .capx
[tester.zip](https://github.com/ChadoriRebornXD/Construct-Master-Collection/files/4173923/tester.zip)

## Steps to Reproduce
1. Add MobileOneSignal Plugin
2. Add C3 GooglePlay Plugin
3. Build Signed APK

## Expected behavior
I expect the build to work.

## Screenshots
![Plugins](https://user-images.githubusercontent.com/34767801/74081828-ddb00080-4a5b-11ea-9ba5-bbb9030c0032.PNG)

## Construct Editor
 - Construct 3

## Construct Runtime
 - Construct 3

 ## Construct 3 Version 
r185

## Additional Info
I assume the play service versions are creating a conflict. But I am not sure. The issue occurs when adding the C3 GooglePlay plugin and/or the Gameanalytics plugin.

Below is the error log shown by C3s build service.
[log.txt](https://github.com/ChadoriRebornXD/Construct-Master-Collection/files/4173930/log.txt)

**Comment by Chadori** [comment]
Thank you for reporting. Please give me time to confirm.

**Comment by Chadori** [comment]
Hi. It seems there is a dependency issue. 
Although, I was able to **build successfully** through the **Cordova CLI** and only had the problem through the **C3 Build Service**.

Would appreciate if you could report this to the **Construct Team** instead.

**Debug:**
```
:app:transformDexArchiveWithExternalLibsDexMergerForDebug
```
```
* What went wrong:
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives:
Learn how to resolve the issue at https:/~~/dependencies#duplicate_classes.
Program type already present: android.support.v4.app.INotificationSideChannel
```
**Release:**
```
:app:transformDexArchiveWithExternalLibsDexMergerForRelease
```
```
* What went wrong:
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForRelease'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives:
Learn how to resolve the issue at https:/~~/dependencies#duplicate_classes.
Program type already present: android.support.v4.app.INotificationSideChannel
```

## Situation
Here's the current situation, based on my tests.

In the **C3 Build Service**, if I mix these plugins:
```
construct-play-games@1.1.2
onesignal-cordova-plugin@2.8.3
```
The build fails.

But, in the **C3 Build Service**, if I mix these plugins:
```
construct-play-games@1.1.2
onesignal-cordova-plugin@2.8.3
construct-mobile-advert@1.5.2
```
The build succeeds. 

Although, with the **Cordova CLI**:
```
construct-play-games@1.1.2
onesignal-cordova-plugin@2.8.3
```
The build succeeds, unlike the C3 Build Service.

-----

## Screenshots

**Google Play + Mobile Advert + OneSignal** [C3 Build Service]

![image](https://user-images.githubusercontent.com/31282960/74102810-04685700-4b82-11ea-83ee-e8fe4cc543fa.png)


**Google Play + OneSignal** [C3 Build Service]

![image](https://user-images.githubusercontent.com/31282960/74102815-12b67300-4b82-11ea-865a-528e6121e2b9.png)


**Google Play + OneSignal** [Cordova CLI]

![image](https://user-images.githubusercontent.com/31282960/74102833-23ff7f80-4b82-11ea-90bc-f72f99e2bf67.png)

**Comment by Chadori** [comment]
This error isn't new to me. I didn't report this, because the Construct Team isn't obliged to support 3rd party addons, in their time for bug fixing. Which I understand so I simply ignored it.

But addons with a Play Services dependency usually don't go along well with Construct 3 objects with  Play Services dependencies too. 

Similar to the **Mobile GooglePlus** addon with 
```
SUCCESS ::
construct-mobile-advert@1.5.2
construct-play-games@1.1.2
cordova-plugin-googleplus@8.2.1
```
&
```
SUCCESS ::
construct-play-games@1.1.2
cordova-plugin-googleplus@8.2.1
```
But,
```
FAILURE ::
construct-mobile-advert@1.5.2
cordova-plugin-googleplus@8.2.1
```

Also, with the same error:
```
app:transformDexArchiveWithExternalLibsDexMergerForRelease
```

-----

I'd be willing to make a topic in the forums to discuss if we could get a solution for this, if you'd help out in convincing them. 🙂 But there's no guarantee if the **Construct Team** has time to help us out and support external addons.

**Comment by Boulerzzz** [comment]
Thank you so much for looking into this! Works great now! 🙂
I was able to build trough C3 build service. I only had to remove GameAnalytics and keep GooglePlay, Mobile Advert & OneSignal. Its nice to know it should work with CLI as well. I'll probably have to build trough CLI later when I want to replace the standard bell with my own small icons for the notifications. 👍 

Also the onesignal website is really nice nowdays. I used onesignal back in the days and they really improved their site since then 👍 

I agree there is a chance Construct team would ignore such topic. After all they are a small team and need to prioritize/maintain their own plugins first.
